wolfcrypt/src/random.c

```diff
@@ -363,14 +361,14 @@ static int Hash_gen(DRBG* drbg, byte* out, word32 outSz, const byte* V)
                 drbg->lastBlock = checkBlock;
             }
 
-            if (out != NULL) {
+            if (out != NULL && outSz != 0) {
                 if (outSz >= OUTPUT_BLOCK_LEN) {
                     XMEMCPY(out, digest, OUTPUT_BLOCK_LEN);
                     outSz -= OUTPUT_BLOCK_LEN;
                     out += OUTPUT_BLOCK_LEN;
                     array_add_one(data, DRBG_SEED_LEN);
                 }
-                else if (out != NULL && outSz != 0) {
+                else if (out != NULL) {
```

Comment:
I'll clean this up further. The issue found here was because this whole section is inside a for loop and outSz is set to 0, which caused a warning with a memcpy. But it was def false positive here.